File: loops/loops.ts
function main() {
  // For loop
  for (let i = 0; i < 5; i++) {
    console.log(i);
  }

  // While loop
  let j = 0;
  while (j < 5) {
    console.log(j);
    j++;
  }

  // Infinite loop
  let k = 0;
  while (true) {
    console.log(k);
    k++;
    if (k === 5) {
      break;
    }
  }

  // For...of loop
  const numbers = [1, 2, 3, 4, 5];
  for (const value of numbers) {
    console.log("Value:", value);
  }

  // forEach loop
  numbers.forEach((value, index) => {
    console.log("Value:", value, "Index:", index);
  });
}

main();

File: loops/loops.go
package main

import "fmt"

func main() {
	// For loop
	for i := 0; i < 5; i++ {
		fmt.Println("For Loop", i)
	}

	// While loop (for loop without initialization and post statement)
	j := 0
	for j < 5 {
		fmt.Println("While Loop no initialization", j)
		j++
	}

	// Infinite loop
	k := 0
	for {
		fmt.Println("Infinity with break", k)
		k++
		if k == 5 {
			break
		}
	}

	//For Range
	numbers := []int{3, 4, 5, 6, 7}
	for i, n := range numbers {
		fmt.Printf("For Range \n Index: %d, Number: %d\n", i, n)
	}

}
